patches/api/0420-Brigadier-based-command-API.patch

```diff
@@ -0,0 +1,746 @@
++        return PROVIDER.player();
++    }
++
++    /**
++     * Represents a selector that can capture multiple
++     * player entities.
++     *
++     * @return argument that takes multiple players
++     */
++    @NotNull
++    public static ArgumentType<ArgumentResolver<Collection<Player>>> players() {
++        return PROVIDER.players();
++    }
++
++    /**
++     * Represents a selector that provides list
++     * of player profiles
++     *
++     * @return player profile arguments
++     */
++    @NotNull
++    public static ArgumentType<ArgumentResolver<Collection<PlayerProfile>>> playerProfiles() {
++        return PROVIDER.playerProfiles();
++    }
++
++    /**
++     * A block position argument.
++     *
++     * @return argument
++     */
++    @NotNull
++    public static ArgumentType<ArgumentResolver<BlockPosition>> blockPos() {
++        return PROVIDER.blockPos();
++    }
++
++    /**
++     * A signed message argument.
++     * This argument can be resolved to retreieve the underlying
++     * signed message.
++     *
++     * @return argument
++     */
++    @NotNull
++    public static ArgumentType<MessageArgumentResponse> signedMessage() {
++        return PROVIDER.signedMessage();
++    }
++
++
++    /**
++     * A component argument.
++     *
++     * @return argument
++     */
++    @NotNull
++    public static ArgumentType<Component> component() {
++        return PROVIDER.component();
++    }
++
++    @ApiStatus.Internal
++    interface VanillaArgumentProvider {
++
++        @NotNull
++        ArgumentType<ItemStack> itemStack();
++
++        @NotNull
++        ArgumentType<BlockState> blockState();
++
++        @NotNull
++        ArgumentType<MessageArgumentResponse> signedMessage();
++
++        @NotNull
++        ArgumentType<UUID> uuid();
++
++        @NotNull
++        ArgumentType<ArgumentResolver<Entity>> entity();
++
++        @NotNull
++        ArgumentType<ArgumentResolver<Player>> player();
++
++        @NotNull
++        ArgumentType<ArgumentResolver<Collection<Entity>>> entities();
++
++        @NotNull
++        ArgumentType<ArgumentResolver<Collection<Player>>> players();
++
++        @NotNull
++        ArgumentType<ArgumentResolver<Collection<PlayerProfile>>> playerProfiles();
++
++        @NotNull
++        ArgumentType<ArgumentResolver<BlockPosition>> blockPos();
++
++        @NotNull
++        ArgumentType<Component> component();
++
++        void validatePrimitiveType(@NotNull ArgumentType<?> type);
++    }
++
++}
+diff --git a/src/main/java/org/bukkit/Bukkit.java b/src/main/java/org/bukkit/Bukkit.java
+index 1ab3d8aee3101ca08d9adf5d9ff9f83de911594c..24fa91a9b54dca3e97e37a9da677c2b5d0e42550 100644
+--- a/src/main/java/org/bukkit/Bukkit.java
++++ b/src/main/java/org/bukkit/Bukkit.java
+@@ -2374,6 +2374,7 @@ public final class Bukkit {
+      * @return the default message
+      */
+     @NotNull
++    @Deprecated // Used only for bukkit commands
```

Comment:
I think `Obsolete` might be better here, but either way you should add that source comment to the javadocs instead.